### ISSUE TITLE
在BNNneck之后测试， 没有之前好？

### ISSUE THREAD

**Issue by chernbo** [michuanhaohao/reid-strong-baseline#165]
非常感谢您和您伙伴的出色工作，这里我发现一个问题，在设置_C.TEST.NECK_FEAT = 'after' 和‘before’时，我发现after 测试结果不如‘before’,如果我没有记错的话，论文里面是测试阶段，应该在BCNNneck之后测试效果更好，但是我得到实验结果相反。我尝试看其他的issues ，但是没有找到类似问题。所以您能帮我解答一下困惑吗？

下面是log记录：
2020-09-02 15:26:04,375 reid_baseline INFO: Using 1 GPUS
2020-09-02 15:26:04,375 reid_baseline INFO: Namespace(config_file='', opts=[])
2020-09-02 15:26:04,375 reid_baseline INFO: Running with config:
DATALOADER:
  NUM_INSTANCE: 16
  NUM_WORKERS: 8
  SAMPLER: softmax
DATASETS:
  NAMES: market1501
  ROOT_DIR: /home/cb/DataSet/
INPUT:
  PADDING: 10
  PIXEL_MEAN: [0.485, 0.456, 0.406]
  PIXEL_STD: [0.229, 0.224, 0.225]
  PROB: 0.5
  RE_PROB: 0.5
  SIZE_TEST: [384, 128]
  SIZE_TRAIN: [384, 128]
MODEL:
  DEVICE: cuda
  DEVICE_ID: 0
  IF_LABELSMOOTH: on
  IF_WITH_CENTER: no
  LAST_STRIDE: 1
  METRIC_LOSS_TYPE: triplet
  NAME: resnet50
  NECK: bnneck
  PRETRAIN_CHOICE: self
  PRETRAIN_PATH: /home/cb/Reid_Model/reid-strong-baseline-master/modeling/trained_weight/market_resnet50_model_120.pth
OUTPUT_DIR: /home/cb/Reid_Model/reid-strong-baseline-master/log/market1501
SOLVER:
  BASE_LR: 0.0003
  BIAS_LR_FACTOR: 2
  CENTER_LOSS_WEIGHT: 0.0005
  CENTER_LR: 0.5
  CHECKPOINT_PERIOD: 10
  CLUSTER_MARGIN: 0.3
  EVAL_PERIOD: 20
  GAMMA: 0.1
  IMS_PER_BATCH: 32
  LOG_PERIOD: 10
  MARGIN: 0.3
  MAX_EPOCHS: 50
  MOMENTUM: 0.9
  OPTIMIZER_NAME: Adam
  RANGE_ALPHA: 0
  RANGE_BETA: 1
  RANGE_K: 2
  RANGE_LOSS_WEIGHT: 1
  RANGE_MARGIN: 0.3
  STEPS: (30, 55)
  WARMUP_FACTOR: 0.3333333333333333
  WARMUP_ITERS: 500
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0
TEST:
  FEAT_NORM: yes
  IMS_PER_BATCH: 128
  NECK_FEAT: after
  RE_RANKING: no
  WEIGHT: /home/cb/Reid_Model/reid-strong-baseline-master/modeling/trained_weight/market_resnet50_model_120.pth
2020-09-02 15:26:11,648 reid_baseline.inference INFO: Enter inferencing
2020-09-02 15:29:00,209 reid_baseline.inference INFO: Validation Results
2020-09-02 15:29:00,209 reid_baseline.inference INFO: mAP: 68.6%
2020-09-02 15:29:00,209 reid_baseline.inference INFO: CMC curve, Rank-1  :85.2%
2020-09-02 15:29:00,209 reid_baseline.inference INFO: CMC curve, Rank-5  :93.1%
2020-09-02 15:29:00,209 reid_baseline.inference INFO: CMC curve, Rank-10 :95.1%
--------------------------------------------------------------------------------------------------------------------------------------------
2020-09-02 10:29:55,016 reid_baseline INFO: Using 1 GPUS
2020-09-02 10:29:55,016 reid_baseline INFO: Namespace(config_file='', opts=[])
2020-09-02 10:29:55,017 reid_baseline INFO: Running with config:
DATALOADER:
  NUM_INSTANCE: 16
  NUM_WORKERS: 8
  SAMPLER: softmax
DATASETS:
  NAMES: market1501
  ROOT_DIR: /home/cb/DataSet/
INPUT:
  PADDING: 10
  PIXEL_MEAN: [0.485, 0.456, 0.406]
  PIXEL_STD: [0.229, 0.224, 0.225]
  PROB: 0.5
  RE_PROB: 0.5
  SIZE_TEST: [384, 128]
  SIZE_TRAIN: [384, 128]
MODEL:
  DEVICE: cuda
  DEVICE_ID: 0
  IF_LABELSMOOTH: on
  IF_WITH_CENTER: no
  LAST_STRIDE: 1
  METRIC_LOSS_TYPE: triplet
  NAME: resnet50
  NECK: bnneck
  PRETRAIN_CHOICE: self
  PRETRAIN_PATH: 
OUTPUT_DIR: /home/cb/Reid_Model/reid-strong-baseline-master/log/market1501
SOLVER:
  BASE_LR: 0.0003
  BIAS_LR_FACTOR: 2
  CENTER_LOSS_WEIGHT: 0.0005
  CENTER_LR: 0.5
  CHECKPOINT_PERIOD: 10
  CLUSTER_MARGIN: 0.3
  EVAL_PERIOD: 20
  GAMMA: 0.1
  IMS_PER_BATCH: 32
  LOG_PERIOD: 10
  MARGIN: 0.3
  MAX_EPOCHS: 50
  MOMENTUM: 0.9
  OPTIMIZER_NAME: Adam
  RANGE_ALPHA: 0
  RANGE_BETA: 1
  RANGE_K: 2
  RANGE_LOSS_WEIGHT: 1
  RANGE_MARGIN: 0.3
  STEPS: (30, 55)
  WARMUP_FACTOR: 0.3333333333333333
  WARMUP_ITERS: 500
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0
TEST:
  FEAT_NORM: yes
  IMS_PER_BATCH: 128
  NECK_FEAT: before
  RE_RANKING: no
  WEIGHT: /home/cb/Reid_Model/reid-strong-baseline-master/modeling/trained_weight/market_resnet50_model_120.pth
2020-09-02 10:30:00,232 reid_baseline.inference INFO: Enter inferencing
2020-09-02 10:32:52,789 reid_baseline.inference INFO: Validation Results
2020-09-02 10:32:52,789 reid_baseline.inference INFO: mAP: 73.0%
2020-09-02 10:32:52,789 reid_baseline.inference INFO: CMC curve, Rank-1  :87.8%
2020-09-02 10:32:52,789 reid_baseline.inference INFO: CMC curve, Rank-5  :94.9%
2020-09-02 10:32:52,789 reid_baseline.inference INFO: CMC curve, Rank-10 :96.7%

**Comment by InstantWindy** [comment]
@chernbo Are you testing resnet50 on the market dataset?  Why did you get 73% map?  Didn't you use the pre trained resnet50 model？

**Comment by chernbo** [comment]
@InstantWindy ,I sure that I use the pre trained resnet50,but it seems some gap with paper said

**Comment by michuanhaohao** [comment]
STEPS: (30, 55)
WARMUP_FACTOR: 0.3333333333333333
WARMUP_ITERS: 500

Your training settings are not same with the paper.

**Comment by chernbo** [comment]
I will check it again,thanks for your suggestion!

**Comment by chernbo** [comment]
@michuanhaohao   I will check it again,thanks for your suggestion!
